tutorials/Inference-TensorFlow-Bert-Model-for-High-Performance-in-ONNX-Runtime.ipynb

```diff
@@ -0,0 +1,211 @@
+    "!wget -O ./bert_op_scripts/BertOnnxModelTF.py https://raw.githubusercontent.com/microsoft/onnxruntime/master/onnxruntime/python/tools/bert/BertOnnxModelTF.py\n",
+    "!wget -O ./bert_op_scripts/BertOnnxModel.py https://raw.githubusercontent.com/microsoft/onnxruntime/master/onnxruntime/python/tools/bert/BertOnnxModel.py\n",
+    "!wget -O ./bert_op_scripts/OnnxModel.py https://raw.githubusercontent.com/microsoft/onnxruntime/master/onnxruntime/python/tools/bert/OnnxModel.py\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Run the `bert_model_optimization.py` with `--framework tensorflow` option to optimize the converted model. Other notable options to use are \n",
+    "- `--gpu_only`: allow half-precision float for better performance.\n",
+    "- `--input_int32`: Use int32 tensors instead of default int64 as input to avoid un-necessary Cast nodes and get better performance.\n",
+    "- `--float16`: Use float16 tensors instead of default float32 as input to enable half precision floats. Recommended for NVidia GPU with Tensor Core like V100 and T4. For older GPUs, float32 is likely faster."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Below are three examples to run bert_model_optimization.py. Choose one according to your needs and adjust --input\n",
+    "# --output path names as necessary.\n",
+    "\n",
+    "# For CPU\n",
+    "!python bert_op_scripts/bert_model_optimization.py --input <bert.onnx> --output <bert_cpu.onnx> --framework tensorflow\n",
+    "\n",
+    "# # For inferences under NVidia GPU with Tensor Core like V100 and T4\n",
+    "# !python bert_op_scripts/bert_model_optimization.py --input <bert.onnx> --output <bert_gpu_fp16.onnx> --framework tensorflow --gpu_only –float16\n",
+    "\n",
+    "# # For inferences under other NVidia GPUs except V100 and T4\n",
+    "# !python bert_op_scripts/bert_model_optimization.py --input <bert.onnx> --output <bert_gpu_fp32.onnx> --framework tensorflow --gpu_only\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3 - Inference the Optimized Model with ONNX Runtime\n",
+    "\n",
+    "#### Install ONNX Runtime\n",
+    "Install the latest ONNX Runtime if you haven't done so already. \n",
+    "\n",
+    "Install one `onnxruntime` python build. Choose to install `onnxruntime` to use CPU features, or `onnxruntime-gpu` to enjoy GPU execution providers such as CUDA. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Install ONNX Runtime for CPU\n",
+    "!{sys.executable} -m pip install -U onnxruntime\n",
+    "\n",
+    "## Alternatively, install onnxruntime for GPU\n",
```

Comment:
in the pytorch tutorial, CPU vs GPU is automatically selected. why not here

Reply:
In the PyTorch notebook, we can auto-select CPU/GPU package because we know whether GPU is used in code (via `torch.cuda.is_available()`). In TensowFlow 1.13 and 1.14, `tensorflow` and `tensorflow-gpu` packages are installed separately. We cannot tell which one is used in the notebook. So we leave the choice to the users here to select the onnxruntime package they'd like to use.

Reply:
In the PyTorch notebook, we can auto-select cpu/gpu package because we know whether gpu is used in code. In Tensorflow 1.13 and 1.14, tensorflow and tensorflow-gpu packages are installed separately. We can't tell which one is used in the notebook. So here, we leave the choice to the users to select the onnxruntime package they'd like to use.